src/haddock/gear/config_reader.py

```diff
@@ -40,32 +46,117 @@
 # thanks https://stackoverflow.com/questions/39158902
 _sub_header_re = re.compile(r'^ *\[(\w+(?:\.\w+)+)\]')
 
-# https://regex101.com/r/q2fuFl/1
-_string_re = re.compile(r'''^ *(\w+) *= *("(.*?)"|'(.*?)')''')
+# string regular expression here matches the expectations of modules' defaults
+# there are no modules defaults that expect any other character. So, if we
+# block it's read from here, avoid also downstream errors.
+# https://regex101.com/r/TRMx0B/1
+_string_re = re.compile(
+    r'''^ *(\w+) *= *'''
+    r'''("([\w\-]*?)"|'([\w\-]*?)')'''
+    )
 
 # https://regex101.com/r/6X4j7n/2
-_number_re = re.compile(r'^ *(\w+) *= *(\-?\d+\.?\d*|\-?\.\d+|\-?\.?\d+[eE]\-?\d+|-?\d+\.?\d*[eE]\d+)(?: |#|$)')  # noqa: #501
+_number_re = re.compile(
+    r'^ *(\w+) *= *'
+    r'(\-?\d+\.?\d*|\-?\.\d+|\-?\.?\d+[eE]\-?\d+|-?\d+\.?\d*[eE]\d+)(?: |#|$)'
+    )
 
 # https://regex101.com/r/K6yXbe/1
 _none_re = re.compile(r'^ *(\w+) *= *([Nn]one|[Nn]ull)')
 
 _nan_re = re.compile(r'^ *(\w+) *= *([nN][aA][nN])')
 
-# https://regex101.com/r/YCZSAo/1
-_list_one_liner_re = re.compile(r'^ *(\w+) *= *(\[.*\])')
+# https://regex101.com/r/YCZSAo/4
+_list_one_liner_re = re.compile(
+    r'''^ *(\w+) *= *(\[[\w\ \,\-\"\'\[\]\.\\\/]*\])'''
+    )
+
+# https://regex101.com/r/bWlaWB/2
+_list_multiliner_re = re.compile(
+    r'''^ *(\w+) *= *\[(?:\ *[\w\ \,\-\"\'\[\]\.\\\/]*'''
+    r'''[\w\ \,\-\"\'\[\.\\\/](?:#+[\w\ ]*)?)?$'''
+    )
 
-# https://regex101.com/r/bWlaWB/1
-_list_multiliner_re = re.compile(r" *(\w+) *= *\[\ *#?[^\]\n]*$")
+_list_multiline_content_re = re.compile(
+    r'''^ *([\w\ \,\-\"\'\[\]\.\\\/]*)'''
+    )
 
 # https://regex101.com/r/kY49lw/1
 _true_re = re.compile(r'^ *(\w+) *= *([tT]rue)')
 _false_re = re.compile(r'^ *(\w+) *= *([fF]alse)')
 
+# https://regex101.com/r/X4i3Je/5
+
+_keys_that_accept_files = [
+    "cns_exec",
+    "executable",
+    "molecules",
+    RUNDIR,
+    ]
+
+_keys_files_regex = "|".join(_keys_that_accept_files)
 
-class _Path_re:
+_file_linux_re = re.compile(
+    r'''^ *(\w+_fname'''
+    f'''|{_keys_files_regex}) *= *'''
+    r'''("((\.{1,2}\/)*[\w\-\/]+[\w\-]\.?[\w\-]+)"'''
+    r'''|'((\.{1,2}\/)*[\w\-\/]+[\w\-]\.?[\w\-]+)')'''
+    )
+
+_file_windows_re = re.compile(
```

Comment:
a bit unclear to me, will it parse things **generated** with windows or **to be executed in windows**? does this mean we are adding windows support?